### PR TITLE
Enable API reporting for protocol-base and protocol-definitions

### PR DESCRIPTION
**Pull request by tylerbutler**
This PR enables API reporting (as [described in our wiki](https://github.com/microsoft/FluidFramework/wiki/API-reports-and-review)) for the protocol-\* packages. Note that this PR alone will not yet cause the build to fail if the API changes. That will be a future change.

### CODE REVIEW

File: server/routerlicious/packages/protocol-definitions/api-extractor.json [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
   "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
-  "extends": "@fluidframework/build-common/api-extractor-common.json"
+  "extends": "@fluidframework/build-common/api-extractor-common-report.json",
+  "docModel": {
+    "enabled": true,
+    "apiJsonFilePath": "<projectFolder>/../../../../_api-extractor-temp/doc-models/<unscopedPackageName>.api.json"
+  },
+  "apiReport": {
+    "enabled": true,
+    "reportFolder": "<projectFolder>/../../../../api-report/",
```
Comment (tylerbutler) [inline server/routerlicious/packages/protocol-definitions/api-extractor.json:10]:
```suggestion
    "reportFolder": "<projectFolder>/../../../../api-report/"
```

**Comment by tylerbutler** [comment]
@tanviraumi @anthony-murphy Can you take a quick look at this? It won't affect the build at this point. I'll fix the merge conflicts before merging.

**Review by anthony-murphy — APPROVED** [review]

**Comment by anthony-murphy** [comment]
looks good to me

**Review by tanviraumi — APPROVED** [review]

**Comment by tanviraumi** [comment]
Looks good!

**Pull request merged**